code/game/mecha/mecha.dm

```diff
@@ -924,6 +938,154 @@
+	//and if they're present try to smash them
+	//Failing to break any object we crash into will return and end execution
+	for(var/obj/obstacle in get_turf(src))
+		if((obstacle.flags & ON_BORDER) && (src != obstacle))
+			if(!obstacle.CheckExit(src, target))
+				brokesomething = 1
+				if (!crash_into(obstacle))
+					done = 1//If it survived the impact then we stop breaking things for this proc
+
+
+
+
+
+
+
+	//3. Now we hit the turf itself, if it's a wall
+	if (!done && !target.CanPass(src, target))
+		crash_into(target)
+		brokesomething = 1
+		if (!target.CanPass(src, target))
+			done = 1
+
+
+
+	//4. Now we search the target tile for any dense objects that also block us.
+	//This could be girders left behind from the wall we just destroyed
+	if (!done)
+		for (var/atom/A in target)
+			if (A.density && A != src && A != occupant && A.loc != src)
+				brokesomething = 1
+				if (!crash_into(A))
+					done = 1//If it survived the impact then we stop breaking things for this proc
+
+
+	if (brokesomething)//If we hit any >0 number of things, whether they broke or not, we play the impact sound exactly once
+		playsound(get_turf(target), 'sound/weapons/heavysmash.ogg', 100, 1)
+
+
+	//5. If we get here, then we've broken through everything that could stop us
+	//Step forward into the tile and display a victory message!
+	//Its also possible to get here if we crashed against something that offered no resistance
+		//like an airlock that opened when bumped
+		//Or a mob/locker that got pushed away
+		//No damage will be taken in this case
+	if (!done && target.Enter(src, null))
+		if (health <= 0)
+			return 0//This prevents bugginess if the exosuit breaks while crashing into stuff
+
+		mechstep(dir)
+		if (brokesomething)
+			src.visible_message("<span class='danger'>[src.name] breaks through!</span>")
+		return
+	else
+		//if we fail to step forward, then we do the attack animation instead
+		target = get_step(src, dir)//re-fetch target just incase
+		do_attack_animation(target)
+
+
+
+/obj/mecha/proc/crash_into(var/atom/A)
+	var/aname = A.name//Cache this mainly because turfs change name when broken
+	var/oldtype = A.type
+	if (health <= 0)
+		return 0//This prevents bugginess if the exosuit explodes/dies while crashing into stuff
+
+
+	var/damage = crash_damage(A)
+
+	A.ex_act(3)
+	sleep(1)
+	if (A && !(A.gcDestroyed) && A.type == oldtype)//We check if the object has been qdel'd or (for turfs) changed type
+		src.visible_message("<span class='danger'>[src.name] crashes into the [aname]</span>")
+		take_damage(damage)
+		return 0//If it survived the impact then we stop breaking things for this proc
+	else
+		take_damage(damage*0.5)//An object that breaks hurts less than one that resists the impact
+
+	return 1
+
+
+/obj/mecha/proc/crash_damage(var/A)
+	if (istype(A, /mob/living))
+		var/mob/living/M = A
+		return min((M.mob_size / 3),2)//Crashing into a cow or cyborg hurts more than crashing into a dog

```

Comment:
The line you've commented here calculates the damage dealt **to the exosuit** when it hits a mob.
Mob damage is decided in that mob's ex_act function. Crashing into a mob just calls ex_act(3) on it

i didn't touch the damage dealt to mobs, but the 25-30 value that showed up in testing seemed fine. That will down someone in ~four hits and break plenty of bones